service/controller/machinepool/handler/migration/machinepool_test.go

```diff
@@ -0,0 +1,294 @@
+package migration
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	oldcapiexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+
+	"github.com/giantswarm/azure-operator/v5/service/unittest"
+)
+
+func TestEnsureCreatedMachinePoolIsCorrect(t *testing.T) {
+	testCases := []struct {
+		name               string
+		oldMachinePoolFile string
+		newMachinePoolFile string
+		expected           expected
+	}{
+		{
+			name:               "case 0: New MachinePool references not updated",
```

Comment:
New MachinePool CR is still does not have Metadata and Spec fully set (references are not updated), so new handler should not do anything yet.